connection/mysql.go

```diff
@@ -702,26 +702,51 @@ func (dbw *DBWrapper) SqlBulkInsert(rows []Row, stmt *BulkInsertStmt) error {
 
 	DbBulkInserts.Inc()
 
-	placeholders := make([]string, len(rows))
-	values := make([]interface{}, len(rows)*stmt.NumField)
-	j := 0
+	done := make(chan struct{})
+	defer close(done)
 
-	for i, r := range rows {
-		placeholders[i] = stmt.Placeholder
+	chErr := make(chan error)
+	wg := sync.WaitGroup{}
+	wgDone := make(chan struct{})
 
-		for _, v := range r.InsertValues() {
-			values[j] = v
-			j++
-		}
-	}
+	for b := range ChunkRows(done, rows, 500) {
+		wg.Add(1)
+		go func(bulk []Row) {
```

Comment:
Does this parallelism even have an effect? Can you insert >1x into the same table concurrently? Doesn't MySQL lock them?

Reply:
CC @lippserd

Reply:
MySQL with InnoDB supports parallel data inserts. It is best used with different tables of course but also works with the very same table. But don't nail me down on how it's done 😆 It could be the case that we face deadlocks here and then. If so, the inserts should be re-executed.

Reply:
OK, then we should keep this behavior at least "just to be sure" not to loose performance.